scipy/io/matlab/mio.py

```diff
@@ -157,6 +157,47 @@ def loadmat(file_name, mdict=None, appendmat=True, **kwargs):
     >>> mat_contents['testdouble']
     array([[0.        , 0.78539816, 1.57079633, 2.35619449, 3.14159265,
             3.92699082, 4.71238898, 5.49778714, 6.28318531]])
+
+    By default SciPy reads MATLAB structures as structured NumPy arrays of
+    dtype object. This can be disabled by setting the optional arguement
+    `struct_as_record=False`.
+
+    Get the filename for an example .mat file that contains a MATLAB struct
+    called `teststruct` and load the contents.
+
+    >>> matstruct_fname = pjoin(data_dir, 'teststruct_7.4_GLNX86.mat')
+    >>> matstruct_contents = sio.loadmat(matstruct_fname)
+    >>> teststruct = matstruct_contents['teststruct']
+    array(
+      [[(
+        ['Rats live on no evil star.'],
+        [[1.4142135623730951, 2.7182818284590455, 3.141592653589793]],
+        [[(1.4142135623730951+1.4142135623730951j),
+          (2.7182818284590455+2.7182818284590455j),
+          (3.141592653589793+3.141592653589793j)]]
+      )]],
+      dtype=[('stringfield', 'O'), ('doublefield', 'O'), ('complexfield', 'O')]
+    )
+
+    The size of the structured array is the size of the MATLAB struct, not the
+    number elements in any particular field. The shape defaults to 2-D unless
+    the optional argument `squeeze_me=True`.
```

Comment:
... in which case processing removes all length 1 dimensions (by the application of `np.squeeze`).

Reply:
OK, I added a similar clarification, but I left out `np.squeeze` even tho it does actually use this in `mio_utils.pyx` it might not in the future? I just felt this was more info than was actually necessary. IMO the user needs to know that length 1 dimensions are dropped from the shape, but not necessarily how? Instead, I explained this further with another added example that shows what happens if you pass the arg `squeeze_me=True`.